src/AbstractSQLRules2SQL.ts

```diff
@@ -864,6 +865,23 @@ const typeRules: Dictionary<MatchFn> = {
 			return '(' + comparators.join(' || ') + ')';
 		}
 	},
+	ConcatenateWithSeparator: (args, indent) => {
+		checkMinArgs('ConcatenateWithSeparator', args, 2);
+		const [separator, ...textParts] = args.map((arg) => {
+			if (!isAbstractSqlQuery(arg)) {
+				throw new SyntaxError(
+					`Expected AbstractSqlQuery array but got ${typeof arg}`,
+				);
+			}
+			return TextValue(arg, indent);
+		});
+		if (engine === Engines.websql) {
+			throw new SyntaxError(
+				'ConcatenateWithSeparator not supported on: ' + engine,
+			);
+		}
+		return `CONCAT_WS(${separator}, ${textParts.join(', ')})`;
```

Comment:
It's not strictly necessary to separate out the separator, eg
```suggestion
		const textParts = args.map((arg) => {
			if (!isAbstractSqlQuery(arg)) {
				throw new SyntaxError(
					`Expected AbstractSqlQuery array but got ${typeof arg}`,
				);
			}
			return TextValue(arg, indent);
		});
		if (engine === Engines.websql) {
			throw new SyntaxError(
				'ConcatenateWithSeparator not supported on: ' + engine,
			);
		}
		return `CONCAT_WS(${textParts.join(', ')})`;
```

Reply:
Fair point